lib/apiservers/engine/backends/container.go

```diff
@@ -476,6 +473,36 @@ func (c *ContainerBackend) ContainerExecStart(ctx context.Context, eid string, s
 		op.Errorf("Failed to start Exec task for container(%s) due to error (%s)", id, err)
 		return err
 	}
+
+	// now that the exec has begun and the streams are attached we need to inspect and then wait.
+	op.Debugf("Waiting for completion: task(%s), container(%s)", eid, name)
+	handle, err := c.Handle(id, name)
+	if err != nil {
+		op.Errorf("Failed to obtain handle during exec start for container(%s) due to error: %s", id, err)
+		return engerr.InternalServerError(err.Error())
+	}
+
+	ec, err := c.containerProxy.InspectTask(op, handle, eid, id)
+	if err != nil {
+		return err
+	}
+
+	if ec.State == "stopped" {
+		return nil
+	}
+
+	for ec.State == "running" {
+		err = c.containerProxy.WaitTask(op, id, name, eid)
+		if err != nil {
+			return err
```

Comment:
Debating if this should be an error exit - consider if we can simply call InspectTask again to determine current state.
We don't want to end up in a tight loop where WaitTask is returning and error and Inspect task is returning a "running" state, so maybe this conservative approach is best for now.

Reply:
I agonized over whether or not to bail out of the call right here. Technically the AttachStreams has returned(for whatever reason) so a task has reached some sort of resolution or attach streams failed outright.